### PR TITLE
Documentation updates for 0.17

### PR DESCRIPTION
**Pull request by benjaoming**
This substantially updates documentation for 0.17:

 - Amends release notes
 - Cleans up documentation structure
 - Adds documentation for the `kalite` command #4326
 - Fixes errors in development environment setup guide

### CODE REVIEW

**Comment by benjaoming** [comment]
Trying to merge this, the next step is to see if it builds on RTD, which hopefully it does.

The changes in here are mostly harmless, however one change introduced in `kalite.cli` does move around some blocks of code without changing any logic.

File: docs/developer_docs/environment.rst [hunk 1]
```diff
@@ -39,36 +47,20 @@ Now, every time you work on your development environment, just remember to switc
...
-online source, then the quickest option to install the latest stable release
-of KA Lite is ``pip install ka-lite` or `pip install ka-lite-static``.
-
-
 Static vs. Dynamic version
 __________________________
 
 Apart from Python itself, KA Lite depends on a couple of python applications,
-mainly from the Django ecology. These applications can be installed in two ways:
+mainly from the Django ecology. These dependencies can be installed in two ways:
```
Comment (radinamatic) [inline docs/developer_docs/environment.rst]: Maybe `Django ecosystem` instead of `ecology`...? :wink:

File: docs/installguide/release_notes.rst [hunk 2]
```diff
@@ -310,15 +341,14 @@ Bug fixes
 General
 ^^^^^^^
 
-* KA Lite is officially supported on OSX 10.11 (El Capitan).
-* We've revamped the language packs into a new format, called content packs.
-  This results in significantly faster startup times across the board.
-.. WARNING::
-   You will have to redownload all your languages to fully support 0.16.
+ * KA Lite is officially supported on OSX 10.11 (El Capitan).
```
Comment (radinamatic) [inline docs/installguide/release_notes.rst]: 0.17 is supported on macOS Sierra 10.12.

Comment (radinamatic) [inline docs/installguide/release_notes.rst]: So maybe add the comment about it above in `New features`, around line 35?

File: docs/developer_docs/environment.rst (hunk 1 above)
Comment (benjaoming) [inline docs/developer_docs/environment.rst]: Sure, sounds better... it wasn't a change introduced in this PR, if you like, you can change it by clicking "Edit on Github", top right: http://ka-lite.readthedocs.io/en/develop/developer_docs/environment.html

File: docs/installguide/release_notes.rst (hunk 2 above)
Comment (benjaoming) [inline docs/installguide/release_notes.rst]: Oh yes, maybe could you also add a PR for that? Would be great to have you take part in docs authoring!

Thanks for reading through it!

Comment (benjaoming) [inline docs/installguide/release_notes.rst]: Added in #5385

File: docs/developer_docs/environment.rst (hunk 1 above)
Comment (benjaoming) [inline docs/developer_docs/environment.rst]: Added in #5385

**Pull request merged**